src/core/compact_object.h

```diff
@@ -536,6 +539,15 @@ class CompactObjectView {
   CompactObj obj_;
 };
 
+class CompactObjTypeConverter {
+ public:
+  static std::string_view ObjTypeToString(CompactObjType type);
+  static CompactObjType ObjTypeFromString(std::string_view sv);
+
+ private:
+  static const std::unordered_map<CompactObjType, std::string_view> kObjTypeToString;
```

Comment:
sorry but I am against using global hash maps for translating this.
See https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables 
for example.

Please avoid  dynamic allocation at global scope if possible and this use case does not justify this. There is nothing wrong in switch/case. you can use a c array with constant length if that helps you.